vendor/bat-native-ads/src/bat/ads/internal/country_subdivision.h

```diff
@@ -0,0 +1,64 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BAT_ADS_INTERNAL_COUNTRY_SUBDIVISION_H_
+#define BAT_ADS_INTERNAL_COUNTRY_SUBDIVISION_H_
+
+#include <stdint.h>
+
+#include <string>
+#include <map>
+
+#include "bat/ads/internal/ads_impl.h"
+#include "bat/ads/internal/retry_timer.h"
+#include "bat/ads/internal/timer.h"
+
+namespace ads {
+
+class AdsImpl;
+
+class CountrySubdivision {
+ public:
+  CountrySubdivision(
+      AdsClient* ads_client);
+
+  ~CountrySubdivision();
+
+  void FetchCountrySubdivisionIfAllowedForRegion();
+
+  std::string GetCountrySubdivision() const;
+
+  bool IsValidSubdivisionCountryCode(
+    const std::string& country_subdivision_code) const;
+
+ private:
+  void BuildUrl();
+
+  // Returns the client's inferred country subdivision (as definied in
```

Comment:
That comment is confusing. This is a `void` method, it doesn't return anything.

Reply:
> I didn't see a filter to prevent WY targeting. Is that because the filter is actually on the Ads Platform side? (i.e. we don't create campaigns with WY-level targeting)

correct WY filter is implemented on server siede as per requirements, @jsecretan can you point to the corresponding repo/section?

> Also, are the two state-selection UI preferences visible to users outside of the USA?

no, there is a conditional in the UI here https://github.com/brave/brave-core/pull/5602/files#diff-f5413005c422f01e178e0e4b2e388de1R81 (same for Android)

> That comment is confusing. This is a void method, it doesn't return anything.

good point, changed it to "Fetches [...]"

Reply:
chaged, gh didn't recalculate https://github.com/brave/brave-core/pull/5602/files#diff-2b8dd5c95bb138082aa12e6833cee528R39